graph-manager-docs/source/schema-reporting-protocol.mdx

```diff
@@ -0,0 +1,159 @@
+    * This initial call **does not** include an `executableSchema` string.
+    * This initial call **does** include an `executableSchemaId` in the `EdgeServerInfo` input. This is the SHA-256 hash of the edge server's schema, represented as a hexadecimal string. 
+    * If this or any other `reportServerInfo` request fails with a non-2xx response from Apollo, the edge server should retry after 20 seconds.
+2. Apollo responds with a `ReportServerInfoResponse` object. This response tells the edge server: 
+    * How many seconds to wait before sending the next `reportServerInfo` request
+    * Whether the next `reportServerInfo` request should include the `executableSchema` that corresponds to the `executableSchemaId` provided in the previous request.
+3. After waiting the specified number of seconds, the edge server executes `reportServerInfo` again. This request includes a [normalized `executableSchema` string](#schema-normalization) if and only if `withExecutableSchema` was `true` in Apollo's most recent `ReportServerInfoResponse`.
+4. Go to step 2.
+
+### Pseudocode example
+
+```
+val info = EdgeServerInfo(..)
+val schema = normalize("type Query { .. }")
+var withSchema = false
+
+function sendReport() {
+  val executableSchema = if (withSchema) null else schema
+  val response = reportServerInfo(info, executableSchema)
+
+  if (response.withExecutableSchema) {
+    withSchema = true
+  }
+
+  setTimeout(sendReport, response.inSeconds)
+}
+
+sendReport()
+```
+
+## Type definitions
+
+The schema reporting protocol uses the following GraphQL types, referred to in [Protocol sequence](#protocol-sequence) above:
+
+```graphql
+# This type's fields are documented below.
+input EdgeServerInfo {
+  bootId: String!
+  executableSchemaId: String!
+  graphVariant: String! = "current"
+  libraryVersion: String
+  platform: String
+  runtimeVersion: String
+  serverId: String
+  userVersion: String
+}
+
+type ReportServerInfoResponse {
+  inSeconds: Int!
+  withExecutableSchema: Boolean!
+}
+
+type Mutation {
+  service(id: ID!): ServiceMutation!
+}
+
+type ServiceMutation {
+  reportServerInfo(
+    "Only sent if previously requested i.e. received ReportServerInfoResponse with withExecutableSchema = true"
+    executableSchema: String,
+    info: EdgeServerInfo!
+  ): ReportServerInfoResponse
+}
+```
+
+## `EdgeServerInfo` fields
+
+### Required fields
+
+| Name  | Type | Description  |
+|---|---|---|
+| `bootId` | `String!` | A randomly generated UUID that's unique for each instance of your edge server. Set this value on server startup (a given value should not persist across restarts). |
+| `executableSchemaId` | `String!` | A unique identifier for the edge server's schema. Should be the hex representation of the schema document's SHA-256 hash. |
+
+### Recommended fields
+
+| Name  | Type | Description  |
+|---|---|---|
+| `graphVariant` | `String!` | The name of the graph variant to register the schema to. The default value is `current`. |
```

Comment:
Technically this _is_ required in the API, but has a default, so I guess it depends on the question 'when we say required, do we mean "required to provide" or "required to be sent"'.

Reply:
Right, I am erring on the side of organizing by "whether or not your code needs to provide a value" as opposed to "whether or not the field is nullable in the schema"

Reply:
There's often a "default" column for use cases like this.